### DISCUSSION TITLE
Shouldn't default mode for flush be true for writeFile and similar methods?

### DISCUSSION THREAD

**Discussion by ackava** [nodejs/help#4642] [Ideas]
Is it possible that `writeFile` may not throw an error but actual file may not be written due to process crash?

In order to Guarente success of `writeFile` we have to manually put `flush` to true for every operation.

Is there any reason why `flush` is false by default?

**Comment by bnoordhuis** [comment]
> Is it possible that writeFile may not throw an error but actual file may not be written due to process crash?

Process crash? No (barring OS bugs). System crash? Yes.

> Is there any reason why flush is false by default?

Performance. `{flush:true}` translates to `fs.fsync(fd)` and is very slow compared to OS writeback.

**Comment by ackava** [comment]
@bnoordhuis 

```
  await writeFile(name, buffer);
  ..
  const buffer = await readFile(name);
```

If readFile appears after awaiting writeFile with, even if the buffer was not flushed due to hard disk being busy, will readFile provide accurate 100% of the time?

**Reply by bnoordhuis** [reply]
Yes
